dali/operators/bbox/bbox_rotate.cc

```diff
@@ -0,0 +1,417 @@
+                    dali::TensorLayout("xyXY"), false)
+    .AddOptionalArg("bbox_normalized",
+                    R"code(Input bounding boxes are in normalized [0,1] format)code", true, false)
+    .AddOptionalArg(
+        "keep_size",
+        R"code(If true, the bounding box output coordinates will assume the image canvas size was also kept 
+(see ``nvidia.dali.fn.rotate``).)code",
+        false, false)
+    .AddOptionalArg("mode",
+                    R"code(Mode of the bounding box transformation. Possible values are:
+
+ - ``expand``: expands the bounding box to definitely enclose the target, but may be larger than rotated target.
+ - ``fixed``: retains the original size of the bounding box, but may be smaller than the rotated target. 
+ - ``halfway``: halfway between the expanded size and the original size.
+
+.. note::
+  Modes other than ``expand`` are not recommended for datasets with high aspect ratio boxes and high rotation angles.
+)code",
+                    dali::string("expand"), false)
+    .AddOptionalArg(
+        "remove_threshold",
+        R"code(Relative area remaining threshold for removing boxes that fall outside the image boundaries
+after rotation. Should be between [0-1] where 0 means no removal even if box lies outside, 1 means removal 
+if any part of the box is outside.)code",
+        0.1f, false);
+
+using Mode = BBoxRotate<CPUBackend>::Mode;
+
+/**
+ * @brief Expands the bounding boxes to all corners x1y1, x2y1, x1y2, x2y2 in flattened xCoords and
+ *        yCoords.
+ *
+ * @note Not sure if I can use the 16-byte aligned version as an assumption.
+ *
+ * @tparam ltrb If true, the input boxes are in left-top-right-bottom format, otherwise in
+ *         xywh format.
+ * @param inBoxes Input bounding boxes.
+ * @param xCoords Output flattened x coordinates of the corners.
+ * @param yCoords Output flattened y coordinates of the corners.
+ * @param numBoxes Number of bounding boxes.
+ */
+template <bool ltrb>
+void expandToAllCorners(dali::span<const vec<4>> inBoxes, dali::span<float> xCoords,
+                        dali::span<float> yCoords) {
+  for (int i = 0; i < inBoxes.size(); ++i) {
+    auto box = inBoxes[i];
+    // Convert to x2y2
+    if constexpr (!ltrb) {
+      box.z += box.x;
+      box.w += box.y;
+    }
+    const auto offset = 4 * i;
+#if defined(__x86_64__) || defined(_M_X64)
+    // Copy X coordinates
+    const __m128 xvec = _mm_setr_ps(box.x, box.z, box.x, box.z);
+    _mm_storeu_ps(&xCoords[offset], xvec);
+    // Copy Y coordinates
+    const __m128 yvec = _mm_setr_ps(box.y, box.y, box.w, box.w);
+    _mm_storeu_ps(&yCoords[offset], yvec);
+#elif defined(__aarch64__)
+    // Copy X coordinates
+    float32x4_t xvec = {box.x, box.z, box.x, box.z};
+    vst1q_f32(&xCoords[offset], xvec);
+    // Copy Y coordinates
+    float32x4_t yvec = {box.y, box.y, box.w, box.w};
+    vst1q_f32(&yCoords[offset], yvec);
+#endif
```

Comment:
For the sake of completeness and readability, please add a plain C++ fallback.

Reply:
Will do, here's a [godbolt link](https://godbolt.org/z/G81qdE1bq) to the reason why I decided to pull out the intrinsics. I assume there is a lot of this problem in the cpu asm.

The ARM is untested because I don't have big arm computer, I assume you run the uni ttests on both arch?

Reply:
Yes we do.
By the way, we have OpenMP SIMD enabled in our builds, so you can try [this approach](https://godbolt.org/z/o74jMqb81),

Reply:
Very nice, I'll sub in that implementaiton, always best to be cross-platform 🤗